duckduckgo/iOS
# Improve iOS build time by replacing file header check with SwiftLint

Task/Issue URL: https://app.asana.com/0/414235014887631/1203880707578250/f
Tech Design URL:
CC: @brindy 

**Description**:

This PR improves the iOS build time by ~13 seconds by replacing the file header check with a SwiftLint rule. Some changes had to happen to make this possible:

1. Some file headers (20 or so) violated the pattern in subtle ways. These have all been changed to make them consistent.
2. There was a directory full of apparently unused source files which was being detected by SwiftLint. These files have all been removed

Three other small cleanup changes happened here:

1. There is an empty file simply named `a` that currently exists in the root of the iOS directory. It has been removed
2. There was a `-copy` variant of an Info.plist file present in the project file, but there was no actual file for it on disk – the Xcode file viewer indicated this in red. It has been removed from the project file
3. SwiftLint had a warning that one of its enabled rules should have been in the `analyzer_rules` section of its config file, so it has been moved

**Before:**
<img width="384" alt="File Header Check Before" src="https://user-images.githubusercontent.com/183774/216232799-e9a64150-5978-4dca-87c6-d209749ee5c4.png">

**After:**
<img width="374" alt="File Header Check After" src="https://user-images.githubusercontent.com/183774/216232813-6f0694e9-9489-4f33-8555-27ad6a958f34.png">



**Steps to test this PR**:
1. Build the app locally and verify that you get no file header warnings
1. Check that there is no 10+ second build step related to listing or file headers
1. Create a new file and build the app, verify that the new file doesn't violate the header rule



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
